crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,308 @@
+/// 2. Disassemble the package bytecode with `aptos disassemble --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Disassemble {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+/// Decompile the Move bytecode pointed to into Move source code.
+///
+/// For example, if you want to decompile an on-chain package `PackName` at account `0x42`:
+/// 1. Download the package with `aptos move download --account 0x42 --package PackName --bytecode`
+/// 2. Decompile the package bytecode with `aptos decompile --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Decompile {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+#[derive(Debug, Clone, Copy, Default)]
+enum BytecodeCommandType {
+    // Need to set a default for technical reasons, but we override it anyway below.
+    #[default]
+    Disassemble,
+    Decompile,
+}
+
+#[derive(Debug, Parser)]
+pub struct BytecodeCommand {
+    /// Set programmatically
+    #[clap(skip)]
+    command_type: BytecodeCommandType,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(long)]
+    pub is_script: bool,
+
+    /// The path to a directory containing Move bytecode files with the extension `.mv`.
+    /// The tool will process all files find in this directory
+    ///
+    /// If present, a source map at the same location ending in `.mvsm` and the source
+    /// file itself ending in`.move` will be processed by the tool.
+    #[clap(long)]
+    pub package_path: Option<PathBuf>,
+
+    /// Alternatively to a package path, path to a single bytecode file which should be processed.
+    #[clap(long)]
+    pub bytecode_path: Option<PathBuf>,
+
+    /// (Optional) Currently only for disassemble: path to a coverage file for the VM in order
+    /// to print trace information in the disassembled output.
+    #[clap(long)]
+    pub code_coverage_path: Option<PathBuf>,
+
+    /// Output directory for the generated file. Defaults to the directory of the
+    /// `path/module.mv` file if not provided. The disassembled output is stored
+    /// at `output_dir/module.mv.asm`, and decompiled output at
+    /// `output_dir/module.mv.move`.
+    #[clap(long, value_parser)]
+    pub(crate) output_dir: Option<PathBuf>,
+
+    #[clap(flatten)]
+    pub(crate) prompt_options: PromptOptions,
+}
+
+#[async_trait]
+impl CliCommand<String> for Disassemble {
+    fn command_name(&self) -> &'static str {
+        "Disassemble"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<String> {
+        self.command.command_type = BytecodeCommandType::Disassemble;
+        self.command.execute().await
+    }
+}
+
+#[async_trait]
+impl CliCommand<String> for Decompile {
+    fn command_name(&self) -> &'static str {
+        "Decompile"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<String> {
+        self.command.command_type = BytecodeCommandType::Decompile;
+        self.command.execute().await
```

Comment:
I think it'd be simpler if `command_type` wasn't a field on `BytecodeCommand`. What if instead you had something like this:

```
self.command.execute(BytecodeCommandType::Decompile).await
```

It is confusing as it is now because it's part of a struct that is otherwise created with clap (in general I think `#[clap(skip)]` is a sign that things should be restructured), which leads to this `default` confusion above as well.

Reply:
Done